quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java

```diff
@@ -48,5 +55,37 @@
         return new ResponseEntity<SignupUserResponse>(signupUserResponse, HttpStatus.CREATED);
 
     }
+
+    /**
+     * This method expose end point user/signin
+     *
+     * @param authorization : password authorization.
+     * @return SigninResponse
+     */
+
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @RequestMapping(method = RequestMethod.POST, path = "/user/signin", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<SigninResponse> signin(@RequestHeader("authorization") final String authorization) throws AuthenticationFailedException {
+
+        String[] splitText = authorization.split("Basic ");
+
+        byte[] decoder = Base64.getDecoder().decode(splitText[1]);
+        String decodedText = new String(decoder);
+        String[] decodedTextArray = decodedText.split(":");
+
+
```

Comment:
It will be good to check of the passed basic authentication string has ':' and send an error. If ':' is not present, we will accessing a null pointer in the next step